### PR TITLE
AS-688: Integration test for delta layer

### PR DESCRIPTION
**Pull request by ansingh7115**
This test was run locally

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala [hunk 1]
```diff
@@ -981,6 +981,13 @@ class RawlsApiSpec extends TestKit(ActorSystem("MySpec")) with AnyFreeSpecLike w
 
   // Retrieves roles with policy emails for bucket acls and checks that service account is set up correctly
   private def getBucketRolesWithEmails(bucketName: GcsBucketName)(implicit patienceConfig: PatienceConfig): List[(String, Set[String])] = {
+    import org.typelevel.log4cats.Logger
+    import org.typelevel.log4cats.slf4j.Slf4jLogger
+    import org.apache.commons.io.IOUtils
+    import cats.effect.IO
+
+    implicit val logger: org.typelevel.log4cats.StructuredLogger[IO] = Slf4jLogger.getLogger[IO]
+
```
Comment (ansingh7115) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala:990]: fallout from updating google2 lib

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala [hunk 2]
```diff
@@ -326,6 +370,30 @@ class SnapshotAPISpec extends AnyFreeSpecLike with Matchers with BeforeAndAfterA
...
+    )
+
+    bqService.use {
+      bq =>
+        for {
+          dataset <- bq.getDataset(datasetName)
+        } yield {
+          dataset
+        }.toList
+    }.unsafeRunSync()
```
Comment (davidangb) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala]: up to you: you could simplify this to:
```
bqService.use(_.getDataset(datasetName)).unsafeRunSync()
```
which returns an `Option[Dataset]`; in the test you could check to see if the Option is nonEmpty. Or, add the `.toList` like you have now and check its size

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala [hunk 3]
```diff
@@ -156,6 +170,36 @@ class SnapshotAPISpec extends AnyFreeSpecLike with Matchers with BeforeAndAfterA
...
+
+          val snapshotName = "snapshotReferenceForAnalysis"
+          // add snapshot reference to the workspace. Under the covers, this creates the workspace in WSM and adds the ref
+          createSnapshotReference(projectName, workspaceName, dataRepoSnapshotId, snapshotName)
+
+          // validate the snapshot was added correctly: list snapshots in Rawls, should return 1, which we just added.
+          // if we can successfully list snapshot references, it means WSM created its copy of the workspace
+          val listResponse = listSnapshotReferences(projectName, workspaceName)
+          val resources = Rawls.parseResponseAs[ResourceList](listResponse).getResources.asScala
+          resources.size shouldBe 1
```
Comment (davidangb) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala]: if you want to be all Scalatest-y: `resources should have size 1`

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala [hunk 4]
```diff
@@ -156,6 +170,36 @@ class SnapshotAPISpec extends AnyFreeSpecLike with Matchers with BeforeAndAfterA
...
+
+          // check that the bq dataset has been created
+          val resourceId = resources.head.getMetadata.getWorkspaceId
+          val dataset = getDataset("deltalayer_forworkspace_" + resourceId.toString.replace('-', '_'), projectName, ownerAuthToken)
+          dataset.size shouldBe 1
```
Comment (davidangb) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala]: Scalatest goodness for size:
```
dataset should have size 1
```
or, if this changes to return an Option:
```
dataset should not be empty
```

**Review by davidangb — APPROVED** [review]
some suggestions inline, all optional

**Review by davidangb — APPROVED** [review]
still :+1: after the latest changes.

Thinking out loud: now that #1475 has merged, we probably want an integration test to check that deleting a workspace also deletes the dataset; that can be in a future PR

**Review by MatthewBemis — APPROVED** [review]

**Pull request merged**